From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Fri, 17 Mar 2023 01:58:24 +0100
Subject: [PATCH] Set size of file info for symlinks to 0

`ostree_raw_file_to_content_stream` will try to read it, causing a
critical warning with GLib 2.76 causing tests to fail.
---
 common/flatpak-utils.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/flatpak-utils.c b/common/flatpak-utils.c
index f28fdc071b1c..b8ad97003877 100644
--- a/common/flatpak-utils.c
+++ b/common/flatpak-utils.c
@@ -5056,6 +5056,7 @@ flatpak_mtree_create_symlink (OstreeRepo         *repo,
 
   g_file_info_set_name (file_info, filename);
   g_file_info_set_file_type (file_info, G_FILE_TYPE_SYMBOLIC_LINK);
+  g_file_info_set_size (file_info, 0);
   g_file_info_set_attribute_uint32 (file_info, "unix::uid", 0);
   g_file_info_set_attribute_uint32 (file_info, "unix::gid", 0);
   g_file_info_set_attribute_uint32 (file_info, "unix::mode", S_IFLNK | 0777);
